src/embind/embind.js

```diff
@@ -595,47 +595,47 @@ var LibraryEmbind = {
         name: name,
         'fromWireType': function(value) {
             var length = HEAPU32[value >> 2];
-            var a = new Array(length);
-            for (var i = 0; i < length; ++i) {
-                a[i] = String.fromCharCode(HEAPU8[value + 4 + i]);
-            }
+
+            var str = Module['UTF8ToString'](value + 4, length);
```

Comment:
This is lower priority than the main feedback, but perhaps decoding UTF-8 with a known length should be its own function.  There are two bits of dynamic dispatch here: The function lookup on Module and checking whether there is a length.  I don't trust JITs very much, but my concerns could be unfounded.  :)